mix.exs

```diff
@@ -15,16 +15,34 @@ try do
   defmodule AntikytheraInstanceExample.Mixfile do
     use Mix.Project
 
+    # Here we strictly enforce Erlang/OTP version during evaluations of `mix.exs`.
+    # Elixir version is also checked using `:elixir` key in `project/0`.
+    # Although these version checks are not mandatory for antikythera instances,
+    # this way an antikythera instance can control Erlang/Elixir versions used in its gear projects.
+    versions =
+      File.read!(Path.join(__DIR__, ".tool-versions"))
+      |> String.split("\n", trim: true)
+      |> Map.new(fn line -> [n, v] = String.split(line, " ", trim: true); {n, v} end)
+    @elixir_version Map.fetch!(versions, "elixir")
+
+    otp_version         = Map.fetch!(versions, "erlang")
+    otp_version_path    = Path.join([:code.root_dir(), "releases", System.otp_release(), "OTP_VERSION"])
+    current_otp_version = File.read!(otp_version_path) |> String.trim_trailing()
+    if current_otp_version != otp_version do
+      Mix.raise("Incorrect Erlang/OTP version! required: '#{otp_version}', used: '#{current_otp_version}'")
+    end
+
     def project() do
       github_url = "https://github.com/access-company/antikythera_instance_example"
+      base_settings = Antikythera.MixCommon.common_project_settings() |> Keyword.replace!(:elixir, @elixir_version)
```

Comment:
One thing popped in my mind is (although it resides in antikythera's `mix_common.exs`, not in this repository):
if we are to hand off elixir version check responsibility to each antikythera instance,
how about removing `:elixir` field from `common_project_settings/0`?
Always requiring `|> Keyword.replace!(:elixir, whatever_version)` part is not very nice to my eyes.
`mix.exs` files generated by `mix new` task come with `:elixir` field by default so it would be more natural to set that field directly, not via replacing the one provided from `common_project_settings/0`.

Reply:
`Antikythera.MixCommon.common_project_settings/0` is used also by gear projects' `mix.exs`, so we have to consider both antikythera instances and gears.
I'm not sure what kind of version requirements these projects should specify (other than the one in antikythera's mix.exs),
but given that they contain code generated by antikythera's macros, the same version requirement as antikythera's seems to be a reasonable default.
In this regard it's OK to include `:elixir` in return value of `Antikythera.MixCommon.common_project_settings/0`.

Treating this as just a cosmetic issue, antikythera instances can (of course) safely specify the version requirement as follows:
```ex
[
  ...
  elixir: @elixir_version,
  ...
] ++ Antikythera.MixCommon.common_project_settings()
```

Reply:
Well, for the sake of "providing default", yes it would be OK to have `:elixir` spec in `common_project_settings/0`.
Let us keep this.